### ISSUE TITLE
[Broken] Twitch livestream with mpv: [ytdl_hook] ERROR: Failed to download m3u8 playlist information: HTTP Error 403: Forbidden

### ISSUE THREAD

**Issue by xfzv** [yt-dlp/yt-dlp#125]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running yt-dlp version **2021.03.01**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
PASTE VERBOSE LOG HERE

```

## Description

- OS: Arch Linux

- `yt-dlp --version`:

```
2021.03.01
```

- `mpv --version`:

```
mpv 0.33.0-dirty Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
FFmpeg library versions:
   libavutil       56.51.100
   libavcodec      58.91.100
   libavformat     58.45.100
   libswscale      5.7.100
   libavfilter     7.85.100
   libswresample   3.7.100
FFmpeg version: n4.3.1
```

- `mpv.conf`:

```
script-opts=ytdl_hook-ytdl_path="/usr/bin/yt-dlp"
```

Whenever I'm trying to play a Twitch livestream with mpv, I'm getting this:

- `mpv -v "https://www.twitch.tv/summit1g"`:

```sh
[cplayer] Command line options: '-v' 'https://www.twitch.tv/summit1g'
[cplayer] mpv 0.33.0-dirty Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
[cplayer]  built on UNKNOWN
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.51.100
[cplayer]    libavcodec      58.91.100
[cplayer]    libavformat     58.45.100
[cplayer]    libswscale      5.7.100
[cplayer]    libavfilter     7.85.100
[cplayer]    libswresample   3.7.100
[cplayer] FFmpeg version: n4.3.1
[cplayer] 
[cplayer] Configuration: /usr/bin/waf configure --prefix=/usr --confdir=/etc/mpv --enable-cdda --enable-dvb --enable-dvdnav --enable-libarchive --enable-libmpv-shared --disable-build-date
[cplayer] List of enabled features: 52arch alsa asm caca cdda cplayer cplugins cuda-hwaccel cuda-interop debug-build drm dvbin dvdnav egl egl-drm egl-helpers egl-x11 ffmpeg ffnvcodec gbm gbm.h gl gl-wayland glibc-thread-name glob glob-posix gpl iconv jack javascript jpeg lcms2 libarchive libass libavdevice libbluray libdl libm libmpv-shared libplacebo librt linux-fstatfs linux-input-event-codes lua memfd_create optimize plain-gl posix posix-or-mingw pthreads pulse rubberband shaderc shaderc-shared stdatomic uchardet vaapi vaapi-drm vaapi-egl vaapi-vulkan vaapi-wayland vaapi-x-egl vaapi-x11 vdpau vt.h vulkan wayland wayland-protocols x11 xv zlib
[cplayer] Reading config file /etc/mpv/encoding-profiles.conf
[cplayer] Applying profile 'default'...
[cplayer] Reading config file /home/username/.config/mpv/mpv.conf
[cplayer] Applying profile 'default'...
[cplayer] Setting option 'script-opts' = 'ytdl_hook-ytdl_path="/usr/bin/yt-dlp"' (flags = 4)
[cplayer] Setting option 'v' = '' (flags = 8)
[ifo_dvdnav] Opening /home/username/.config/mpv/input.conf
[bdmv/bluray] Opening /home/username/.config/mpv/input.conf
[file] Opening /home/username/.config/mpv/input.conf
[input] Parsing input config file /home/username/.config/mpv/input.conf
[input] Input config file /home/username/.config/mpv/input.conf parsed: 23 binds
[cplayer] Waiting for scripts...
[osd/libass] Shaper: FriBidi 1.0.10 (SIMPLE) HarfBuzz-ng 2.7.4 (COMPLEX)
[osd/libass] Setting up fonts...
[progressbar] Opened config file script-opts/torque-progressbar/main.conf.
[osd/libass] Using font provider fontconfig
[osd/libass] Done.
[cplayer] Set property: shared-script-properties -> 1
[cplayer] Done loading scripts.
[cplayer] Running hook: ytdl_hook/on_load
[ytdl_hook] ytdl:// hook
[ytdl_hook] not a ytdl:// url
[ffmpeg] Opening https://www.twitch.tv/summit1g
[cplayer] Set property: shared-script-properties -> 1
[osd/libass] Shaper: FriBidi 1.0.10 (SIMPLE) HarfBuzz-ng 2.7.4 (COMPLEX)
[osd/libass] Setting up fonts...
[osd/libass] Using font provider fontconfig
[osd/libass] Done.
[cplayer] Set property: shared-script-properties -> 1
[ffmpeg] Mime-type: 'text/html'
[demux] Trying demuxers for level=normal.
[lavf] No format found, try lowering probescore or forcing the format.
[demux] Trying demuxers for level=unsafe.
[lavf] No format found, try lowering probescore or forcing the format.
[cplayer] Opening failed or was aborted: https://www.twitch.tv/summit1g
[cplayer] Running hook: ytdl_hook/on_load_fail
[ytdl_hook] full hook
[ytdl_hook] found youtube-dl at: /usr/bin/yt-dlp
[ytdl_hook] ERROR: Failed to download m3u8 playlist information: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
[ytdl_hook] Traceback (most recent call last):
[ytdl_hook]   File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
[ytdl_hook]     return _run_code(code, main_globals, None,
[ytdl_hook]   File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
[ytdl_hook]     exec(code, run_globals)
[ytdl_hook]   File "/usr/bin/yt-dlp/__main__.py", line 19, in <module>
[ytdl_hook]   File "/usr/bin/yt-dlp/yt_dlp/__init__.py", line 606, in main
[ytdl_hook]   File "/usr/bin/yt-dlp/yt_dlp/__init__.py", line 596, in _real_main
[ytdl_hook]   File "/usr/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2503, in download
[ytdl_hook]   File "/usr/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2568, in post_extract
[ytdl_hook]   File "/usr/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 2555, in actual_post_extract
[ytdl_hook] AttributeError: 'NoneType' object has no attribute 'get'
[ytdl_hook] youtube-dl failed: unexpected error occurred
[cplayer] finished playback, unrecognized file format (reason 4)
[cplayer] Failed to recognize file format.
[cplayer]
[cplayer] Exiting... (Errors when loading file)
[cplayer] Set property: shared-script-properties -> 1
```

It seems that `yt-dlp` cannot detect any format for Twitch livestream, unlike `youtube-dl`:

- `youtube-dl -F "https://www.twitch.tv/summit1g"`:

```
[twitch:stream] summit1g: Downloading stream GraphQL
[twitch:stream] summit1g: Downloading stream access token GraphQL
[twitch:stream] 41875779838: Downloading m3u8 information
[info] Available formats for 41875779838:
format code      extension  resolution note
audio_only       mp4        audio only  163k , mp4a.40.2
160p             mp4        284x160     230k , avc1.4D401F, 30.0fps, mp4a.40.2
360p             mp4        640x360     630k , avc1.4D401F, 30.0fps, mp4a.40.2
480p             mp4        852x480    1431k , avc1.4D401F, 30.0fps, mp4a.40.2
720p60           mp4        1280x720   3426k , avc1.4D401F, 60.0fps, mp4a.40.2
936p60__source_  mp4        1664x936   9037k , avc1.64002A, 60.0fps, mp4a.40.2 (best)
```

- `yt-dlp -F "https://www.twitch.tv/summit1g"`:

```
[twitch:stream] summit1g: Downloading stream GraphQL
[twitch:stream] summit1g: Downloading stream access token GraphQL
[twitch:stream] 41875779838: Downloading m3u8 information
ERROR: Failed to download m3u8 playlist information: HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
```

It works fine when using `youtube-dl` instead of `yt-dlp`:

- `mpv "https://www.twitch.tv/summit1g"`:

```
(+) Video --vid=1 (h264 1920x1080)
 (+) Audio --aid=1 (aac 2ch 44100Hz)
[progressbar] Playing: "summit1g (live) 2021-03-01 11:03"
AO: [pulse] 44100Hz stereo 2ch float
VO: [gpu] 1920x1080 yuv420p
AV: 00:00:05 / 00:00:07 (64%) A-V:  0.000 Cache: 2.4s/1MB
```

**Comment by pukkandan** [comment]
Seems to be a bug introduced by https://github.com/yt-dlp/yt-dlp/commit/310c2ed2c648f4e1254e27d0e8901d629828b2ba

Untill we manage to fix it, you can use [2021.02.19](https://github.com/yt-dlp/yt-dlp/releases/tag/2021.02.19)

**Comment by xfzv** [comment]
> Seems to be a bug introduced by [310c2ed](https://github.com/yt-dlp/yt-dlp/commit/310c2ed2c648f4e1254e27d0e8901d629828b2ba)
> 
> Untill we manage to fix it, you can use [2021.02.19](https://github.com/yt-dlp/yt-dlp/releases/tag/2021.02.19)

Can confirm it works fine with [2021.02.19](https://github.com/yt-dlp/yt-dlp/releases/tag/2021.02.19)

Cheers

**Comment by shirt-dev** [comment]
Issue was caused by not passing HTTP headers to the playlist request. Fixed as of b3b30a4bca45bd99c59475c2b4da88f95b646576.
